### PR TITLE
FDS-205 implemented access control

### PR DESCRIPTION
**Pull request by Baraulia**

### CODE REVIEW

File: handler/user_handlers.go [hunk 1]
```diff
@@ -19,6 +20,12 @@ import (
 // @Failure 500 {object} model.ErrorResponse
 // @Router /users/{id} [get]
 func (h *Handler) getUser(ctx *gin.Context) {
+	necessaryRole := "Superadmin"
+	if err := h.service.CheckRoleRights(nil, necessaryRole, ctx.GetString("perms"), ctx.GetString("role")); err != nil {
+		h.logger.Warnf("Handler getRoleById:not enough rights")
```
Comment (TonySinero) [inline handler/user_handlers.go]: In the description for the logger, all handlers have the same line)))

**Review by Nalivayko13 — COMMENTED** [review]
LGTM

File: handler/user_handlers.go (hunk 1 above)
Comment (Baraulia) [reply handler/user_handlers.go]: fixed

File: handler/router.go [hunk 2]
```diff
@@ -24,18 +23,24 @@ func (h *Handler) InitRoutes() *gin.Engine {
...
-		user.PUT("/", h.updateUser)
-		user.GET("/", h.getUsers)
-		user.POST("/staff", h.createStaff)
-		user.POST("/customer", h.createCustomer)
-		user.POST("/login", h.authUser)
+		userNoAuth.PUT("/", h.updateUser)
+		userNoAuth.POST("/login", h.authUser)
+		userNoAuth.POST("/customer", h.createCustomer)
+		userNoAuth.DELETE("/:id", h.deleteUserByID)
+
```
Comment (Ann40a) [inline handler/router.go]: remove line

File: handler/router.go [hunk 3]
```diff
@@ -24,18 +23,24 @@ func (h *Handler) InitRoutes() *gin.Engine {
...
-		user.DELETE("/:id", h.deleteUserByID)
-		user.PUT("/", h.updateUser)
-		user.GET("/", h.getUsers)
-		user.POST("/staff", h.createStaff)
-		user.POST("/customer", h.createCustomer)
-		user.POST("/login", h.authUser)
+		userNoAuth.PUT("/", h.updateUser)
+		userNoAuth.POST("/login", h.authUser)
+		userNoAuth.POST("/customer", h.createCustomer)
+		userNoAuth.DELETE("/:id", h.deleteUserByID)
```
Comment (Ann40a) [inline handler/router.go]: why updating and deleting user are anonymous routes?

**Review by Ann40a — APPROVED** [review]
Same comments like in previous PR as some of the methods were copied

**Comment by TimofeiShorokhov** [comment]
Approved

File: handler/router.go (hunk 3 above)
Comment (Baraulia) [reply handler/router.go]: fixed it

File: handler/router.go (hunk 2 above)
Comment (Baraulia) [reply handler/router.go]: I fixed it

**Pull request merged**